### PR TITLE
Feat: Implemented verify_certificate function

### LINKED ISSUE #2: Implement verify_certificate smart contract function
Implement a Cairo smart contract that allows employers and institutions to verify the authenticity of academic certificates. This contract will verify the certificate's metadata and the university's digital signature, ensuring that only valid credentials are acknowledged.

## Requirements
Implement a function that checks if a certificate is valid using the parse argument; `certificate_id` and `hashed_key`.
  - The function should check if `certificate_id` & `hashed_key` are valid in the same data object
  - It should check the status of the certificate using `isActive`; if `true` function returns true else emit an event with this description `Certificate has been revoked` and return false
  - The function should return a boolean value if the above returns true.
  - Write test cases should pass
  - Emit event

## Acceptance Criteria

1. The smart contract includes a function that returns a boolean indicating certificate validity.
2. Certificate metadata and the university’s signature are accurately verified.
3. Unit tests cover valid certificates, invalid certificates, and edge cases (e.g., missing data).
4. Code quality and documentation meet the project standards.
5. Event emiited properly

### PR DESCRIPTION
**Pull request by caxtonacollins**
# closes #2 

#### Certificate Verification:
- Added a function to verify the validity of a certificate using its certificate_id and hashed_key. The function checks if the certificate exists, if the hash matches, and if the certificate is active.
-  It returns a boolean and emits an event if the certificate has been revoked.

#### Certificate Revocation:
- Implemented a function that allows the issuing university to revoke a certificate. This sets the certificate’s isActive status to false and emits a CertificateRevoked event.
#### Interface Updates:
- Updated the contract interface to include the new verify_certificate and revoke_certificate functions.
#### Unit Tests:
- Added comprehensive tests for certificate verification, revocation, and issuer-based certificate retrieval, including edge cases.

Checklist
- [x] Certificate verification logic implemented
- [x] Certificate revocation logic implemented
- [x] Events emitted as required
- [x] Interface updated
- [x] Unit tests added and passing

### CODE REVIEW

File: src/certiva.cairo [hunk 1]
```diff
@@ -227,5 +234,46 @@ pub mod Certiva {
...
+                    .emit(
+                        Event::CertificateRevoked(
+                            CertificateRevoked { certificate_id: certificate_id, reason: reason },
+                        ),
+                    );
+                return false;
+            }
+        }
+
+        fn revoke_certificate(ref self: ContractState, certificate_id: felt252) {
```
Comment (JayWebtech) [inline src/certiva.cairo:261]: Please this is not needed remove it

File: src/Interfaces/ICertiva.cairo [hunk 2]
```diff
@@ -30,4 +30,8 @@ pub trait ICertiva<TContractState> {
 
     fn get_certificate(self: @TContractState, certificate_id: felt252) -> Certificate;
     fn get_certicate_by_issuer(ref self: TContractState) -> Array<Certificate>;
+    fn verify_certificate(
+        ref self: TContractState, certificate_id: felt252, hashed_key: ByteArray,
+    ) -> bool;
+    fn revoke_certificate(ref self: TContractState, certificate_id: felt252);
```
Comment (JayWebtech) [inline src/Interfaces/ICertiva.cairo:36]: also this remove it

**Review by JayWebtech — CHANGES_REQUESTED** [review]
